### PR TITLE
[GKD] add ULD type loss to GKD Trainer

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

Add ULD loss to GKD trainer that doesn't require the student/teacher to have the same vocab by optimal transporting the prob from the teacher to student

### CODE REVIEW

File: trl/trainer/gkd_trainer.py [hunk 1]
```diff
@@ -223,25 +287,37 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
 
         # compute teacher output in eval mode
         self.teacher_model.eval()
+        teacher_input_ids = inputs.get("teacher_input_ids", inputs["input_ids"])
+        teacher_attention_mask = inputs.get("teacher_attention_mask", inputs["attention_mask"])
         with torch.no_grad():
             outputs_teacher = self.teacher_model(
-                input_ids=inputs["input_ids"],
```
Comment (qgallouedec) [inline trl/trainer/gkd_trainer.py:228]: Why do you use a different input ids here?

**Comment by qgallouedec** [comment]
Looks good overall. Feel free to request a final review from me when you think it's ready to be merged

File: examples/scripts/gkd.py [hunk 2]
```diff
@@ -42,6 +42,21 @@
     --use_peft \
     --lora_r 64 \
     --lora_alpha 16
+
+# ULD
+python examples/scripts/gkd.py \
```
Comment (qgallouedec) [inline examples/scripts/gkd.py:47]: Where is the argument related to ULD?

**Comment by qgallouedec** [comment]
I'm closing this PR because no recent activity. Please feel free to reopen it or request a reopening if necessary.
